elasticsearch/src/main/java/org/hibernate/search/elasticsearch/analyzer/impl/ElasticsearchAnalyzerStrategy.java

```diff
@@ -14,24 +14,66 @@
 import org.hibernate.search.analyzer.spi.AnalyzerReference;
 import org.hibernate.search.analyzer.spi.AnalyzerStrategy;
 import org.hibernate.search.annotations.AnalyzerDef;
+import org.hibernate.search.cfg.spi.SearchConfiguration;
+import org.hibernate.search.elasticsearch.analyzer.definition.impl.ElasticsearchAnalysisDefinitionRegistryBuilderImpl;
+import org.hibernate.search.elasticsearch.analyzer.definition.spi.ElasticsearchAnalysisDefinitionProvider;
+import org.hibernate.search.elasticsearch.cfg.ElasticsearchEnvironment;
+import org.hibernate.search.elasticsearch.logging.impl.Log;
 import org.hibernate.search.elasticsearch.settings.impl.model.AnalyzerDefinition;
 import org.hibernate.search.elasticsearch.settings.impl.model.CharFilterDefinition;
 import org.hibernate.search.elasticsearch.settings.impl.model.TokenFilterDefinition;
 import org.hibernate.search.elasticsearch.settings.impl.model.TokenizerDefinition;
 import org.hibernate.search.elasticsearch.settings.impl.translation.ElasticsearchAnalyzerDefinitionTranslator;
 import org.hibernate.search.engine.service.spi.ServiceManager;
 import org.hibernate.search.engine.service.spi.ServiceReference;
+import org.hibernate.search.exception.SearchException;
+import org.hibernate.search.util.impl.ClassLoaderHelper;
+import org.hibernate.search.util.impl.ReflectionHelper;
+import org.hibernate.search.util.logging.impl.LoggerFactory;
 
 
 /**
  * @author Yoann Rodiere
  */
 public class ElasticsearchAnalyzerStrategy implements AnalyzerStrategy {
 
+	private static final Log LOG = LoggerFactory.make( Log.class );
+
 	private final ServiceManager serviceManager;
 
-	public ElasticsearchAnalyzerStrategy(ServiceManager serviceManager) {
+	private final SearchConfiguration cfg;
+
+	public ElasticsearchAnalyzerStrategy(ServiceManager serviceManager, SearchConfiguration cfg) {
 		this.serviceManager = serviceManager;
+		this.cfg = cfg;
+	}
+
+	private ElasticsearchAnalysisDefinitionRegistry createDefaultDefinitionRegistry() {
+		ElasticsearchAnalysisDefinitionRegistryBuilderImpl builder =
+				new ElasticsearchAnalysisDefinitionRegistryBuilderImpl();
+
+		String providerClassName = cfg.getProperty( ElasticsearchEnvironment.ANALYZER_DEFINITION_PROVIDER );
+		if ( providerClassName != null ) {
+			ElasticsearchAnalysisDefinitionProvider provider;
+			try {
+				Class<?> providerClazz = ClassLoaderHelper.classForName( providerClassName, serviceManager );
```

Comment:
interesting idea, I wasn't thinking of `@Factory`. The checks I was thinking of are the ones provided by 

- org.hibernate.search.util.impl.ClassLoaderHelper.verifySuperTypeCompatibility(Class<T>, Object, Class<?>, String)
- org.hibernate.search.util.impl.ClassLoaderHelper.checkHasNoArgConstructor(Class<?>, String)
- org.hibernate.search.util.impl.ClassLoaderHelper.checkClassType(Class<?>, String)

I guess you can leave the `@Factory`, it looks good to have it. We probably should unify the two approaches in a follow-up improvement.